service/operator/service.go

```diff
@@ -387,23 +402,25 @@ func (s *Service) GetDeleteState(obj, currentState, desiredState interface{}) (i
 		{
 			newData := map[string]string{}
 			for k, v := range deleteState.ConfigMap.Data {
-				if inConfigMapData(dState.ConfigMapData, k, v) {
+				if !inConfigMapData(dState.ConfigMapData, k, v) {
```

Comment:
Ok. What worries me is that we'll have a lot of nuances in state and it may be difficult to track.

Can you describe how this framework can be applied to following code?

```
err := storage.Delete(KeyFromTPO(obj))
if err != nil && !storage.IsNotExist(err) {
    return err
}
```